rides-android/src/main/java/com/uber/sdk/android/rides/RequestDeeplink.java

```diff
@@ -23,188 +23,23 @@
-            checkNotNull(sessionConfiguration, "Must supply a Login Configuration");
-            checkNotNull(sessionConfiguration.getClientId(), "Must supply client Id on Login Configuration");
-
-
-            Uri.Builder builder = new Uri.Builder();
-            builder.scheme(AppProtocol.DEEPLINK_SCHEME);
-            builder.appendQueryParameter(ACTION, SET_PICKUP);
-            builder.appendQueryParameter(CLIENT_ID, sessionConfiguration.getClientId());
-            if (rideParameters.getProductId() != null) {
-                builder.appendQueryParameter(PRODUCT_ID, rideParameters.getProductId());
-            }
-            if (rideParameters.getPickupLatitude() != null && rideParameters.getPickupLongitude() != null) {
-                addLocation(LocationType.PICKUP, Double.toString(rideParameters.getPickupLatitude()),
-                        Double.toString(rideParameters.getPickupLongitude()), rideParameters.getPickupNickname(),
-                        rideParameters.getPickupAddress(), builder);
-            }
-            if (rideParameters.isPickupMyLocation()) {
-                builder.appendQueryParameter(LocationType.PICKUP.getUriQueryKey(), MY_LOCATION);
-            }
-            if (rideParameters.getDropoffLatitude() != null && rideParameters.getDropoffLongitude() != null) {
-                addLocation(LocationType.DROPOFF, Double.toString(rideParameters.getDropoffLatitude()),
-                        Double.toString(rideParameters.getDropoffLongitude()), rideParameters.getDropoffNickname(),
-                        rideParameters.getDropoffAddress(), builder);
-            }
-
-            String userAgent = rideParameters.getUserAgent();
-            if (userAgent == null) {
-                userAgent = USER_AGENT_DEEPLINK;
-            }
-            builder.appendQueryParameter(USER_AGENT, userAgent);
-
-            return new RequestDeeplink(context, builder.build());
-        }
-
-        private void addLocation(
-                @NonNull LocationType locationType, @NonNull String latitude,
-                @NonNull String longitude, @Nullable String nickname, @Nullable String address, Uri.Builder builder) {
-            String typeQueryKey = locationType.getUriQueryKey();
-            builder.appendQueryParameter(typeQueryKey + LATITUDE, latitude);
-            builder.appendQueryParameter(typeQueryKey + LONGITUDE, longitude);
-            if (nickname != null) {
-                builder.appendQueryParameter(typeQueryKey + NICKNAME, nickname);
-            }
-            if (address != null) {
-                builder.appendQueryParameter(typeQueryKey + FORMATTED_ADDRESS, address);
-            }
-        }
-    }
-
-    private enum LocationType {
-        PICKUP,
-        DROPOFF;
-
-        private String getUriQueryKey() {
-            return name().toLowerCase();
-        }
+@Deprecated
+public class RequestDeeplink extends RideRequestDeeplink {
```

Comment:
Confused on why you're exposing a new public init that's deprecated.

Reply:
The class was previously named RequestDeeplink, I renamed the impl class to RideRequestDeeplink and extended it with the old name to allow for existing customers not to break.